### PR TITLE
chore: refactored docker images replication module

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*

*Description of changes:*
- fixed the deletion workflow and provided an input to remove the hardcoded `idf` as project name
- refactored the shell script to check the repo first, check the image if exists and push images only if needed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by malachi-constant — APPROVED** [review]

**Review by malachi-constant — DISMISSED** [review]
Looks good, are we now defferring to CONTIRBUTING, LICENSE etc at the root of repo?

And looks like there's some static checks to fix

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: fbc0c42245c76658aae1beb71fc033903408560c
* Result: FAILED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c791cace-76d3-40e9-b3bd-5e07442b722f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: a2cc0fc77d05de2f13213c73fce9159f20432ea5
* Result: FAILED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7562e9d6-2066-454d-9d5f-5ed24c4bcd40%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 403c037f028b8130c848ab52a3da288e9e0475a4
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d32ab1d6-6b91-4d6b-b969-6bc17c51ad49%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by srinivasreddych** [comment]
> Looks good, are we now defferring to CONTIRBUTING, LICENSE etc at the root of repo?
> 
> And looks like there's some static checks to fix

Fixed the static checks. Also, from my knowledge, we always the license, contributing.md at the root of the repo. right? I've checked the IDF modules and thats what i observed. Do you have any suggestion?

File: modules/replication/dockerimage-replication/deployspec.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ deploy:
             fi
             sleep 15
           fi
-        - python3 get-list-of-eks-images.py --eks-version ${SEEDFARMER_PARAMETER_EKS_VERSION} --versions-directory data/eks_dockerimage-replication/versions --update-helm-repos --registry-prefix "${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/${AWS_CODESEEDER_NAME}-"
+        - python3 get-list-of-eks-images.py --eks-version ${SEEDFARMER_PARAMETER_EKS_VERSION} --versions-directory data/eks_dockerimage-replication/versions --update-helm-repos --registry-prefix "${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/${SEEDFARMER_PROJECT_NAME}-"        
         - chmod +x replication.sh
```
Comment (kukushking) [inline modules/replication/dockerimage-replication/deployspec.yaml:23]: Nit (not related to change) - is this chmod required? Can we check-in the file with the correct permissions?

File: data/eks_dockerimage-replication/versions/default.yaml [hunk 2]
```diff
@@ -11,6 +11,7 @@ charts:
         tag:
           location: chart
           path: appVersion
+          prefix: v
```
Comment (kukushking) [inline data/eks_dockerimage-replication/versions/default.yaml:14]: Just confirming if this is intentional

**Review by kukushking — APPROVED** [review]
Looks good, thank you!

**Pull request merged**